packages/fetch/src/index.ts

```diff
@@ -236,17 +259,26 @@ export type ${responseTypeName} = ${compositeName} & {
     ${fetchBodyOption}
   }
 `;
+  const throwOnErrorImplementation = `if (!${isNdJson ? 'stream' : 'res'}.ok) {
+    const err: globalThis.Error & {info?: ${hasError ? `${errorName}${override.fetch.includeHttpResponseReturnType ? "['data']" : ''}` : 'any'}, status?: number} = new globalThis.Error();
```

Comment:
Does it make sense to derive this from a generic error type? So the library consumer can for example write a wrapper that handles all API errors?